hibernate-core/src/main/java/org/hibernate/bytecode/enhance/internal/bytebuddy/EnhancerImpl.java

```diff
@@ -433,71 +525,66 @@ private boolean checkUnsupportedAttributeNaming(TypeDescription managedCtClass)
+				// We only need to check this for AccessType.PROPERTY
 				continue;
 			}
 
-			String methodName = methodDescription.getActualName();
-			if (methodName.equals("") ||
-					(!methodName.startsWith("get") && !methodName.startsWith("set") && !methodName.startsWith("is"))) {
-				continue;
-			}
+			final String methodName = methodDescription.getActualName();
 			String methodFieldName;
-			if (methodName.startsWith("is")) { // skip past "is"
-				methodFieldName = methodName.substring(2);
-			}
-			else if (methodName.startsWith("get") ||
-					methodName.startsWith("set")) { // skip past "get" or "set"
-				methodFieldName = methodName.substring(3);
+			if ( methodName.startsWith( "get" ) || methodName.startsWith( "set" ) ) {
+				methodFieldName = methodName.substring( 3 );
 			}
 			else {
-				// not a property accessor method so ignore it
-				continue;
+				assert methodName.startsWith( "is" );
+				methodFieldName = methodName.substring( 2 );
 			}
-			boolean propertyNameMatchesFieldName = false;
-			// extract the property name from method name
-			methodFieldName = getJavaBeansFieldName(methodFieldName);
-			TypeList typeList = methodDescription.getDeclaredAnnotations().asTypeList();
-			if (typeList.stream().anyMatch(typeDefinitions ->
-					(typeDefinitions.getName().equals("jakarta.persistence.Transient")))) {
-				// transient property so ignore it
-				continue;
-			}
-			if (typeList.stream().anyMatch(typeDefinitions ->
-					(typeDefinitions.getName().contains("jakarta.persistence")))) {
-				propertyHasAnnotation = true;
-			}
-			for (FieldDescription ctField : methodDescription.getDeclaringType().getDeclaredFields()) {
-				if (!Modifier.isStatic(ctField.getModifiers())) {
-					AnnotatedFieldDescription annotatedField = new AnnotatedFieldDescription(enhancementContext, ctField);
-					if (enhancementContext.isPersistentField(annotatedField)) {
-						if (methodFieldName.equals(ctField.getActualName())) {
-							propertyNameMatchesFieldName = true;
-							break;
+			// convert first field letter to lower case
+			methodFieldName = getJavaBeansFieldName( methodFieldName );
+			if ( methodFieldName != null && hasPersistenceAnnotation( methodDescription.getDeclaredAnnotations() ) ) {
+				boolean propertyNameMatchesFieldName = false;
+				for ( final FieldDescription field : methodDescription.getDeclaringType().getDeclaredFields() ) {
+					if ( !Modifier.isStatic( field.getModifiers() ) ) {
+						final AnnotatedFieldDescription annotatedField = new AnnotatedFieldDescription(
+								enhancementContext,
+								field
+						);
+						if ( enhancementContext.isPersistentField( annotatedField ) ) {
+							if ( methodFieldName.equals( field.getActualName() ) ) {
+								propertyNameMatchesFieldName = true;
+								break;
+							}
 						}
 					}
 				}
-			}
-			if ( propertyHasAnnotation && !propertyNameMatchesFieldName ) {
```

Comment:
I find myself reading https://jakarta.ee/specifications/persistence/3.2/jakarta-persistence-spec-3.2#a113 again for checking this section.  My question here is whether we can really remove the check if any method has a Persistence annotation?  With the current code in this change determines the access type is PROPERTY than I guess its not FIELD.  I'd like to have more confidence but we should move forward with this change and correct later if needed.

Reply:
I don't understand the question.

Reply:
Pasting from spec:

> The explicit access type may be overridden at the attribute level. That is, a class which explicitly specifies an access type using the Access annotation may also have fields or properties annotated Access, and so the class may have a mix of access types.
>
>  - When Access(FIELD) is specified at the class level, an individual attribute within the class may be selectively designated for property access by annotating a property getter Access(PROPERTY). Mapping annotations for this attribute must be placed on the getter. If a mapping annotation is placed on a property getter which is not annotated Access(PROPERTY), the behavior is undefined.
> - When Access(PROPERTY) is specified at the class level, an individual attribute within the class may be selectively designated for field access by annotating an instance variable Access(FIELD). Mapping annotations for this attribute must be placed on the field. If a mapping annotation is placed on a field which is not annotated Access(FIELD), the behavior is undefined.

Reply:
I ran some local Jakarta EE 10 TCK tests and I'm seeing a TCK failure reported by https://hibernate.atlassian.net/browse/HHH-18928:
> [javatest.batch] FAILED........com/sun/ts/tests/jpa/core/metamodelapi/attribute/Client.java#getJavaMember_from_appmanaged
[javatest.batch] FAILED........com/sun/ts/tests/jpa/core/metamodelapi/attribute/Client.java#getJavaMember_from_appmanagedNoTx
[javatest.batch] FAILED........com/sun/ts/tests/jpa/core/metamodelapi/attribute/Client.java#getJavaMember_from_pmservlet
[javatest.batch] FAILED........com/sun/ts/tests/jpa/core/metamodelapi/attribute/Client.java#getJavaMember_from_puservlet
[javatest.batch] FAILED........com/sun/ts/tests/jpa/core/metamodelapi/attribute/Client.java#getJavaMember_from_stateful3
[javatest.batch] FAILED........com/sun/ts/tests/jpa/core/metamodelapi/attribute/Client.java#getJavaMember_from_stateless3

In HHH-18928 I mentioned:
> It looks like we are enhancing https://github.com/jakartaee/platform-tck/blob/10.0.x/src/com/sun/ts/tests/jpa/core/metamodelapi/attribute/Order.java but I think the default access type for Order entity is PROPERTY so with this change shouldn't we not enhance https://github.com/jakartaee/platform-tck/blob/10.0.x/src/com/sun/ts/tests/jpa/core/metamodelapi/attribute/Order.java?  

Note that https://github.com/jakartaee/platform-tck/blob/10.0.x/src/com/sun/ts/tests/jpa/core/metamodelapi/attribute/Order.java#L47 has an ID annotation so the default access type should be PROPERTY:
```
  @Id
  public int getId() {
    return id;
  }
```

From test output for https://github.com/jakartaee/platform-tck/blob/10.0.x/src/com/sun/ts/tests/jpa/core/metamodelapi/attribute/Client.java#L248C16-L248C36:

> [javatest.batch] 12-12-2024 11:28:24:  SVR-TRACE: attribute JavaMember = total 
[javatest.batch] 12-12-2024 11:28:24:  SVR-ERROR: Expected: getTotal, actual:total

I'm not 100% sure if this is a bug or should be a challenge.  

From the TCK description of the test assertion:

```
        <assertion required="true" impl-spec="false" status="active" testable="true">
            <id>PERSISTENCE:JAVADOC:1214</id>
            <description>Return the java.lang.reflect.Member for the represented attribute.</description>
            <package>jakarta.persistence.metamodel</package>
            <class-interface>Attribute</class-interface> 
            <method name="getJavaMember" return-type="java.lang.reflect.Member"/>
        </assertion>
```

Reply:
@scottmarlow right, Order has an `@Id` annotation so it should default to `AccessType.PROPERTY`. However, the TCK test seems to assert the java member returned by the metamodel, so I do not think it's related with the bytecode enhancement process at all; maybe there's an issue somewhere else in how Hibernate determines which member to return.

You can try creating a test in Hibernate with the same entity mapping and the same assertions, and see if it fails. Then, we can try to identify the cause of this behavior.

Reply:
> @scottmarlow right, Order has an `@Id` annotation so it should default to `AccessType.PROPERTY`. However, the TCK test seems to assert the java member returned by the metamodel, so I do not think it's related with the bytecode enhancement process at all; maybe there's an issue somewhere else in how Hibernate determines which member to return.

For what it is worth, the TCK test passes if the Order class is not enhanced.  With my https://github.com/hibernate/hibernate-orm/pull/9405 change, I had added an `else if access type == PROPERTY` check that would ensure we do not enhance the Order class.  Of course, this also caused around 20-30 test failures as well (I forget the exact count) so my change was incomplete.  Paste of code from that PR:

```
if ( propertyHasAnnotation && !propertyNameMatchesFieldName ) {
				switch ( strategy ) {
					case SKIP:
@@ -501,11 +528,71 @@ else if (methodName.startsWith("get") ||
						// We shouldn't even be in this method if using LEGACY, see top of this method.
						throw new AssertionFailure( "Unexpected strategy at this point: " + strategy );
				}
			} else if ( propertyHasAnnotation && (defaultAccessType == null || AccessType.PROPERTY.equals( defaultAccessType ) ) ) {
```

Also worth noting that the TCK test is calling https://jakarta.ee/specifications/persistence/3.2/apidocs/jakarta.persistence/jakarta/persistence/metamodel/attribute#getJavaMember() which returns a https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/lang/reflect/Member.html so Java reflection is involved and we need to understand how/why that returns JavaMember that returns the wrong name from the [JavaMember.getName()](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/lang/reflect/Member.html#getName()) call.  

> 
> You can try creating a test in Hibernate with the same entity mapping and the same assertions, and see if it fails. Then, we can try to identify the cause of this behavior.

Is there an existing Hibernate ORM test that combines metamodel + bytecode enhancement that we could update to also do what this TCK does?